code/datums/components/storage/storage.dm

```diff
@@ -233,7 +233,7 @@
 
 /datum/component/storage/proc/quick_empty(mob/M)
 	var/atom/A = parent
-	if((!ishuman(M) && (A.loc != M)) || (M.stat != CONSCIOUS) || M.incapacitated())
+	if(M.canUseStorage() && A.Adjacent(M) && (A.loc != M) && M.incapacitated())

```

Comment:
loc != M is cheaper and stronger check than adjacent (and practically the adjacent is redundant but you feel free to keep it in), so that should go first.

Reply:
gotcha! ty

Reply:
Also, what do you mean "add these for AIs as well"? I already have a mob/camera canUseStorage() proc